server/src/main/scala/io/delta/kernelsharedtable/DeltaSharedTableKernel.scala

```diff
@@ -285,13 +303,51 @@ class DeltaSharedTableKernel(
     val protocol = getResponseProtocol(snapshot.protocol, respondedFormat)
 
     val metadata = getResponseMetadata(snapshot.metadata, version = null, respondedFormat)
-    val actions = Seq(
+    var actions = Seq(
       protocol,
       metadata
     )
 
     if (includeFiles) {
-      throw new DeltaSharingUnsupportedOperationException("not implemented yet")
+      if (predicateHints.isEmpty && jsonPredicateHints.isEmpty && limitHint.isEmpty) {
+        val predicateOpt = Option.empty
+        val scanBuilder = snapshot.kernelSnapshot.getScanBuilder(engine)
+        val scan = predicateOpt match {
```

Comment:
will `predicateOpt` be a different value? do we need this `match` here?

Reply:
it wont be now since we arent doing any json predicate hints. I thought to keep this for the skeleton for when we do filtering from json hints. I can remove it now if you want

Reply:
it's preferred to only add what's needed in the PR.

Reply:
fixed